sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1001,16 +1005,18 @@ private boolean isManagementToken(String lockToken) {
                 logger.info("{}: Management node Update completed. Disposition: {}. Lock: {}.",
                     entityPath, dispositionStatus, lockToken);
 
+                message.setIsSettled(true);
```

Comment:
If the message is already settled, we should throw error with message something like `The message has either been deleted or already settled `

Check line item 2 on this.

https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/15/Doc.aspx?sourcedoc={41b0af1e-8da3-4d88-b296-9b616c3e7b6b}&action=edit&wd=target%28Design.one%7C14d18aa9-3941-cf4d-b2fa-4b211ac30733%2FException%5C%2FError%20handling%7Ce696adb4-72c6-4a64-85fb-1d187557099b%2F%29&wdorigin=703

Reply:
Throwing error message now.